fnet/cli/train_model.py

```diff
@@ -55,26 +54,17 @@ def init_cuda(gpu: int) -> None:
         logger.exception('Failed to init CUDA')
 
 
-def get_dataloader_train(
-        args: argparse.Namespace,
-        n_iter_remaining: int,
-) -> torch.utils.data.DataLoader:
-    """Creates DataLoader for training."""
-    bpds_kwargs = copy.deepcopy(args.bpds_kwargs)
-    bpds_kwargs['npatches'] = n_iter_remaining*args.batch_size
+def get_bpds_train(args: argparse.Namespace):
+    """Creates data provider for training."""
     ds_fn = str_to_object(args.dataset_train)
     if not isinstance(ds_fn, Callable):
         raise ValueError('Dataset function should be Callable')
     ds = ds_fn(**args.dataset_train_kwargs)
-    bpds = fnet.data.BufferedPatchDataset(dataset=ds, **bpds_kwargs)
-    dataloader = torch.utils.data.DataLoader(bpds, batch_size=args.batch_size)
-    return dataloader
+    return fnet.data.BufferedPatchDataset(dataset=ds, **args.bpds_kwargs)
 
 
-def get_dataloader_val(
-        args: argparse.Namespace
-) -> Optional[torch.utils.data.DataLoader]:
-    """Creates DataLoader for validation."""
+def get_bpds_val(args: argparse.Namespace):
```

Comment:
Nit: no return type annotation on this function

Also without knowing too much as to what is happening, is it really best here to return optionally None? Why not throw an error? If the dataset val is None then I feel like a `NoDatasetError` or something should be thrown?

Reply:
I pushed an update to add some return type annotations. For training models, providing a validation dataset is optional, so this function returns either the validation dataset or None. The script checks the type later downstream.